buildstrap

```diff
@@ -11,18 +11,42 @@ OUT_TMP=$4
 PACKAGES=$5
 INSTALL_BCC=$6
 MIRROR=$7
+DISTRO_VERSION=$8
 
-root=sudo
+if [[ ! -z $DISTRO_VERSION ]]; then
+    # Assume non-empty DISTRO_VERSION is Ubuntu; see if there's a base image..
+    echo "Checking for Ubuntu Base image from the web..."; echo ""
+    if [ ! -f $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz ]; then
+        curl -L http://cdimage.ubuntu.com/ubuntu-base/releases/$DISTRO_VERSION/release/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz \
+            --output $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz; fi
+    if [ -f $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz ]; then
+        time tar -C $OUT_TMP --exclude=dev -xf $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz;
+        mkdir $OUT_TMP/dev; fi; fi
 
-if [ $EUID -ne 0 ]; then echo "The next stage runs as sudo, please enter password if asked."; fi
+if [ ! -f $OUT_TMP/etc/debian_version ]; then
+    root=sudo
 
-# It's easier to just really be root for debootstrap
-time $root qemu-debootstrap --arch $ARCH --include="$(make_csv "$PACKAGES")" \
-	$DISTRO $OUT_TMP $MIRROR
+    if [ $EUID -ne 0 ]; then echo "The next stage runs as sudo, please enter password if asked."; fi
 
-# Some reason debootstrap leaves these mounted
-$root umount $OUT_TMP/proc/sys/fs/binfmt_misc || true
-$root umount $OUT_TMP/proc || true
+    # It's easier to just really be root for debootstrap
+    time $root qemu-debootstrap --arch $ARCH --include="$(make_csv "$PACKAGES")" \
+        $DISTRO $OUT_TMP $MIRROR
+
+    # Some reason debootstrap leaves these mounted
+    $root umount $OUT_TMP/proc/sys/fs/binfmt_misc || true
+    $root umount $OUT_TMP/proc || true
+else
```

Comment:
Since these if/else blocks are so huge, I suggest split them into 2 different functions. Maybe `do_debian` and `do_ubuntu` and then call them from the if/else.